tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringTests.cs

```diff
@@ -0,0 +1,207 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Telemetry;
+using Datadog.Trace.TestHelpers;
+using Datadog.Trace.TestHelpers.DataStreamsMonitoring;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using VerifyTests;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests;
+
+[UsesVerify]
+[Collection(nameof(KafkaTests.KafkaTestsCollection))]
+[Trait("RequiresDockerDependency", "true")]
+public class DataStreamsMonitoringTests : TestHelper
+{
+    public DataStreamsMonitoringTests(ITestOutputHelper output)
+        : base("DataStreams.Kafka", output)
+    {
+        SetServiceVersion("1.0.0");
+        EnableDebugMode();
+    }
+
+    /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+    /// <summary>
+    /// This sample does a series of produces and consumes to create two pipelines:
+    ///  - service -> topic 1 -> Consumer 1 -> topic 2 -> Consumer 2 -> topic 3 -> consumer 3
+    ///  - service -> topic 2 -> Consumer 2 -> topic 3 -> consumer 3
+    /// Each node (apart from 'service') in the pipelines above have a unique hash
+    ///
+    /// In mermaid (view at https://mermaid.live/), this looks like:
+    /// sequenceDiagram
+    ///    participant A as Root Service<br>(12926600137239154356)
+    ///    participant T1 as Topic 1<br>(2704081292861755358)
+    ///    participant C1 as Consumer 1<br>(5289074475783863123)
+    ///    participant T2a as Topic 2<br>(2821413369272395429)
+    ///    participant C2a as Consumer 2<br>(9753735904472423641)
+    ///    participant T3a as Topic 3<br>(5363062531028060751)
+    ///    participant T2 as Topic 2<br>(246622801349204431)
+    ///    participant C2 as Consumer 2<br>(3398817358352474903)
+    ///    participant T3 as Topic 3<br>(16689539899325095461 )
+    ///
+    ///    A->>+T1: Produce
+    ///    T1-->>-C1: Consume
+    ///    C1->>+T2a: Produce
+    ///    T2a-->>-C2a: Consume
+    ///    C2a->>+T3a: Produce
+    ///
+    ///    A->>+T2: Produce
+    ///    T2-->>-C2: Consume
+    ///    C2->>+T3: Produce
+    /// </summary>
+    [SkippableFact]
+    [Trait("Category", "EndToEnd")]
+    [Trait("Category", "ArmUnsupported")]
+    public async Task SubmitsDataStreams()
+    {
+        SetEnvironmentVariable(ConfigurationKeys.DataStreamsMonitoring.Enabled, "1");
+
+        using var agent = EnvironmentHelper.GetMockAgent(useTelemetry: true);
+
+        using var processResult = RunSampleAndWaitForExit(agent);
+
+        using var assertionScope = new AssertionScope();
+        var dataStreams = agent.DataStreams;
```

Comment:
Is there a reason you didn't invoke `agent.WaitForDataStreams()`?

Reply:
Um, I forgot I added it 😅